### DISCUSSION TITLE
2.7.0

### DISCUSSION THREAD

**Discussion by ruibaby** [halo-dev/halo#4160] [General]
## 新特性

- 支持页面静态缓存（默认关闭）。 by @JohnNiang in <https://github.com/halo-dev/halo/pull/4091>
- 访客发表评论支持限流。 by @JustinLiang522 in <https://github.com/halo-dev/halo/pull/4084>
- 增加登录失败次数限制功能。 by @JohnNiang in <https://github.com/halo-dev/halo/pull/4062>
- 对评论回复添加频率限制。 by @JohnNiang in <https://github.com/halo-dev/halo/pull/4135>
- 限制注册接口的请求速率。 by @JohnNiang in <https://github.com/halo-dev/halo/pull/4128>
- 默认编辑器支持不通过附件库直接插入图片、视频、音频。 by @ruibaby in <https://github.com/halo-dev/halo/pull/4126>

## 功能优化

- 优化 Console 端部分页面复制功能的兼容性问题。 by @ruibaby in <https://github.com/halo-dev/halo/pull/4114>
- Console 端仪表盘的内置小组件添加权限设置。 by @ruibaby in <https://github.com/halo-dev/halo/pull/4112>
- Console 端默认类型的按钮添加鼠标悬浮的样式。 by @ruibaby in <https://github.com/halo-dev/halo/pull/4108>
- 优化 Console 端弹框组件（VModal）在 iOS 设备的高度问题。 by @ruibaby in <https://github.com/halo-dev/halo/pull/4106>
- 维护 Console 端的开发基础设施。 by @ruibaby in <https://github.com/halo-dev/halo/pull/4105>
- 移除 Console 端页面的 lang 属性，避免出现浏览器的翻译提示。 by @longjuan in <https://github.com/halo-dev/halo/pull/4085>
- 优化 Console 端登录页面的三方登录方式按钮，禁用重复点击。 by @ruibaby in <https://github.com/halo-dev/halo/pull/4066>
- Console 端三方登录支持重定向参数。 by @ruibaby in <https://github.com/halo-dev/halo/pull/4063>
- Console 端的文章分类支持在新建时指定上级分类。 by @ruibaby in <https://github.com/halo-dev/halo/pull/4056>

## 问题修复

- 修复 Console 端的下拉框组件选项在移动端无法正常点击的问题。 by @ruibaby in <https://github.com/halo-dev/halo/pull/4116>
- 修复开发环境下访问 Console 出现异常的问题。 by @JohnNiang in <https://github.com/halo-dev/halo/pull/4096>
- 修复主题和插件静态资源的路径遍历漏洞。 by @guqing in <https://github.com/halo-dev/halo/pull/4072>
- 修复登录时出现用户名不存在的问题。 by @JohnNiang in <https://github.com/halo-dev/halo/pull/4070>
- 修复 AttachmentReconciler 重复执行的问题。 by @Contour-D in <https://github.com/halo-dev/halo/pull/4052>
- 修复 Meta Description 中含有特殊字符导致页面显示异常的问题。 by @Contour-D in <https://github.com/halo-dev/halo/pull/4031>
- 修复插件重启后 MainClass 对象缓存未清除的问题。 by @guqing in <https://github.com/halo-dev/halo/pull/4023>
- 修复文章页 Meta Description 标签重复问题。 by @guqing in <https://github.com/halo-dev/halo/pull/4069>

## 依赖更新

- 升级 Console 端 FormKit 的依赖版本至 0.17.3。 by @ruibaby in <https://github.com/halo-dev/halo/pull/4103>
- 升级 Google Guava 版本至 32.0.1-jre。 by @guqing in <https://github.com/halo-dev/halo/pull/4081>
- 升级 Spring Boot 至 3.1.1。 by @JohnNiang in <https://github.com/halo-dev/halo/pull/4136>

## 开发者相关

- Console 端的默认编辑器支持被扩展。 by @ruibaby in <https://github.com/halo-dev/halo/pull/4090>
- 插件的 Controllers 支持自定义 API Group。 by @guqing in <https://github.com/halo-dev/halo/pull/4065>
- 统一主题和插件描述文件的部分字段。 by @guqing in <https://github.com/halo-dev/halo/pull/4061>
- 支持在插件中使用 @ConfigurationProperties 注解。 by @JohnNiang in <https://github.com/halo-dev/halo/pull/4043>
- Console 端的评论来源显示支持通过插件扩展。 by @ruibaby in <https://github.com/halo-dev/halo/pull/4039>
- 为主题端的文章和自定义页面内容添加扩展点。 by @guqing in <https://github.com/halo-dev/halo/pull/4080>
- 优化主题端评论的 Finder API，list 方法支持查询所有来源的评论。 by @guqing in <https://github.com/halo-dev/halo/pull/4104>

## 文档

<https://docs.halo.run>

## 资源

适用于 2.0 的插件和主题可查阅：

- <https://halo.run/store/apps>
- <https://github.com/halo-sigs/awesome-halo>

**Full Changelog**: https://github.com/halo-dev/halo/compare/v2.6.0...v2.7.0

<hr /><em>This discussion was created from the release <a href='https://github.com/halo-dev/halo/releases/tag/v2.7.0'>2.7.0</a>.</em>

**Comment by mobeicanyue** [comment]
鸡汤来喽 开发者大大辛苦了:grin:
